nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -441,11 +551,33 @@ in {
         nextcloud-setup = let
           c = cfg.config;
           writePhpArrary = a: "[${concatMapStringsSep "," (val: ''"${toString val}"'') a}]";
+          requiresReadSecretFunction = c.dbpassFile != null
+            || (c.objectstore != null && c.objectstore.s3.enable);
+          objectstoreConfig = let
+            class = if c.objectstore.s3.enable then "S3" else "";
+            args = if c.objectstore.s3.enable then c.objectstore.s3.arguments else {};
```

Comment:
Nix is lazily evaluated. As soon as you drop the unnecessary submodule above and work with a single `enable` option, you can use e.g. `s3.arguments` (which should default to `{}` then rather than `null`) here without this if-else. I guess the same applies to `class =`.